hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/filtermanagement/TargetFilterAddUpdateLayout.java

```diff
@@ -135,9 +157,46 @@ public ComponentContainer getRootComponent() {
         return filterQueryLayout;
     }
 
+    protected void setCSVExportCallback(final ProxyTargetFilterQuery proxyEntity){
+        StreamResource csvResource = getStreamResourceToCSV(proxyEntity);
+        fileDownloader = new FileDownloader(csvResource);
+        fileDownloader.extend(exportButton);
+    }
+
+    protected void removeDownloaderExtension(){
+        if(fileDownloader != null){
+            fileDownloader.remove();
+        }
+    }
+
+    private StreamResource getStreamResourceToCSV(final ProxyTargetFilterQuery proxyEntity){
+
+        final StreamSource streamSource = () -> {
+            final PipedInputStream in = new PipedInputStream();
+
+            try {
+                final PipedOutputStream out = new PipedOutputStream(in);
+                final TargetDataCsvExporter csvExporter = new TargetDataCsvExporter(targetManagement, ";",
+                        "E MMM dd HH:mm:ss z yyyy");
+
+                SpringContextHolder.getInstance().getBean("uiExecutor", Executor.class)
+                        .execute(() -> csvExporter.writeTargetsByFilterId(proxyEntity.getId(), out));
+
+            } catch (final IOException ex) {
+                LOG.warn("Creating piped Stream failed: ", ex);
+            }
+            return in;
+        };
+
+        return new StreamResource(streamSource, proxyEntity.getName() + ".csv");
+    }
+
     private void addValueChangeListeners() {
         searchButton.addClickListener(event -> onSearchIconClick());
-        autoCompleteComponent.addValidationListener((valid, message) -> searchButton.setEnabled(valid));
+        autoCompleteComponent.addValidationListener((valid, message) -> {
+            searchButton.setEnabled(valid);
+            exportButton.setEnabled(valid);
```

Comment:
Check if we need to additionally verify if name is present before activating the button

Reply:
All good suggestions - thanks! 

By name, are you referring to the fileName of CSV export here? If so, we do not need the fileName to be present for activating the button. The button only needs decoration elements, e.g., caption/label and tooltip.